pkg/storegateway/blocks.gohtml

```diff
@@ -90,5 +103,82 @@
     {{ end }}
     </tbody>
 </table>
+<script>
+    // Get the dropdown and elements to control
+    var dropdown = document.getElementById("action-type");
+    var elementsToControl = document.querySelectorAll(".item-to-control");
+
+    // Function to toggle visibility for each element
+    function toggleVisibility(isVisible) {
+        elementsToControl.forEach(function (element) {
+            if (isVisible) {
+                element.removeAttribute("hidden");
+            } else {
+                element.setAttribute("hidden", "hidden");
+            }
+        });
+    }
+
+    // Add an event listener to the dropdown
+    dropdown.addEventListener("change", function () {
+        var isVisible = dropdown.value !== "none";
+        console.log("Setting visibility to: " + isVisible + " for " + dropdown.value);
+        toggleVisibility(isVisible);
+    });
+
+    // Set the initial visibility based on the selected option
+    var isVisible = dropdown.value !== "none";
+    toggleVisibility(isVisible);
+
+    // Function to reload the page with the action
+    function reloadWithAction() {
+        var form = document.getElementById('configForm');
+
+        // Get the blocks uid from the table and add it to the form
+        var table = document.getElementById('blocksInfo');
+
+        // Create an array to hold the structured table data
+        var blockUIDs = [];
+
+        // Iterate through the table rows
+        var rows = table.getElementsByTagName('tr');
+        for (var i = 1; i < rows.length; i++) { // Start from 1 to skip header row
+            var cells = rows[i].getElementsByTagName('td');
+
+            // Extract uid from the "Block ID" column (assuming it's the second column)
+            var checkboxStatus = cells[0].getElementsByTagName('input')[0].checked;
+            if (!checkboxStatus) {
+                continue;
+            }
+            // Add the uid to the array
+            blockUIDs.push(cells[1].textContent);
+        }
+
+        // Check if the array is empty
+        if (blockUIDs.length === 0) {
+            // Display an alert popup if the array is empty
+            alert("No block seleted for action!");
+        }
+
+        // Check if the input field already exists
+        var blockUIDsField = form.querySelector('input[name="block_uids"]');
```

Comment:
technically we call them ULIDs or IDs, but not UIDs. Let's stick to ULIDs, wdyt?